src/lib/link-meta/link-meta.ts

```diff
@@ -62,8 +63,9 @@ export async function getLinkMeta(
     const controller = new AbortController()
     const to = setTimeout(() => controller.abort(), timeout || 5e3)
 
+    const serciceUrl = await agent.getServiceUrl()
     const response = await fetch(
-      `${LINK_META_PROXY(agent.service.toString() || '')}${encodeURIComponent(
+      `${LINK_META_PROXY(serciceUrl.toString() || '')}${encodeURIComponent(
```

Comment:
Is there a way we could do this without having to have access to the `.getServiceUrl` ? This is the only place this method (from `@atproto/api`) is needed for.